File: JavaScript/7.js
// Declare a variable and assign it a string value. Then print that string
// following this rule: only the letters c o d e r should be included in your 
// output if they are present in the word. Otherwise, exclude that specific 
// letter. An example would be: 'condition' would be printed as 'codo'.
// (40pts)

// 1. Declare a variable and assign it a string value

let str = "condition";
let answer = "";
str = str.split("")

for (let char of str) {
    if (char == "c" || char == "o" || char == "d" || char == "e" || char == "r") {
        answer += char;
    }
}

console.log(answer)


// Q: Is the current letter c, o, d, e, r ??

// C - YES
// O - YES
// N - NO
// D - YES
// I - NO
// T - NO
// I - NO
// O - YES
// N - NO

File: Ruby/7.rb
# Declare a variable and assign it to a string. The code should print "Yes" if
# all the words in the string begin with an uppercase letter. Print "No"
# otherwise. You cannot use a library for this solution.
# (40 pts)

str = "Hello World This Is My Sentence"

def all_words_caps?(str)
  str.split.all? do |word|
    word[0] == word[0].upcase
  end
end

arr = [1, 3, 5, 7, 9]

def all_odd?(arr)
  arr.all? do |n|
    n.odd?
  end
end

puts all_odd?(arr)